### DISCUSSION TITLE
How can I get Meson to build an Objective-C++ project using Meson?

### DISCUSSION THREAD

**Discussion by dreamer-coding** [mesonbuild/meson#13081] [Q&A]
How can I get Meson to build an Objective-C++ project? Secondly is the support for Objective-C++ tested on all common platforms like Windows and Posix-based host systems? Lastly are there possibly undocumented features such as `objcpp_std` flag as an example?

**Comment by dcbaker** [comment]
ObjC++ works as far as our test cases go, if you find issues please report them.

All you need to build ObjC++ is:
```meson
project('myproject', 'objcpp')

<targets includings .mm sources>
```

Yes,  there is an `objcpp_std` option

**Reply by dreamer-coding** [reply]
Is there also a objc_std for the Objective-C side of things?

**Reply by dcbaker** [reply]
Yes

**Reply by dcbaker** [reply]
There may be some supported standards missing

**Reply by dreamer-coding** [reply]
Got an issue when trying to compile Objective-c++ with the Foundation framework on a POSIX system already tried any possible solutions beyond GNUStep and including the GNUA Objective C++ compiler: ![image](https://github.com/mesonbuild/meson/assets/55331536/7a394541-af23-4c5f-956b-ebe158fa918a)
